src/meta-srv/src/weighted.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use rand::distributions::WeightedIndex;
+use rand::prelude::Distribution;
+use rand::rngs::OsRng;
+
+/// A common trait for weighted balance algorithm.
+pub trait WeightedChoose<Item>: Send + Sync {
```

Comment:
Can we move weigited.rs into the selector’s dir? Because it seems that only selector depends on it